beangulp/importer.py

```diff
@@ -128,3 +128,55 @@ def file_date(self, file) -> Optional[date]:
           (If no date is returned, the file creation time is used. This is the
           default.)
         """
+
+
+class Importer:
+
+    @property
+    def name(self) -> str:
+        return f'{self.__class__.__module__}.{self.__class__.__name__}'
+
+    def identify(self, path: str) -> bool:
+        return False
+
+    def extract(self, path: str, existing: data.Entries = None) -> data.Entries:
+        raise NotImplementedError
+
+    # file() support interface
+
+    def account(self, path: str) -> data.Account:
+        raise NotImplementedError
+
+    def date(self, path: str) -> Optional[date]:
+        raise NotImplementedError
+
+    def filename(self, path: str) -> Optional[str]:
+        raise NotImplementedError
+
+
+class Adapter(Importer):
```

Comment:
Please add a liberal docstring describing what this is for.
(Might be obvious to you and I, but still.)